app/features-json/build_json_controller.rb

```diff
@@ -92,6 +84,47 @@ class BuildJSONController < APIController
       json(build_summary_view_model)
     end
 
+    get "#{HOME}/:build_number/logs" do |project_id, build_number|
+      # `current_build_runner` is only defined if the build was just run a while back
+      # if the server was restarted, we're gonna end here in this code block
+      build_log_artifact = current_build.artifacts.find do |current_artifact|
+        # We can improve the detection in the future, to actually mark an artifact as "default output"
+        current_artifact.type.include?("log") && current_artifact.reference.end_with?("fastlane.log")
+      end
+
+      if build_log_artifact
+        artifact_file_content = File.read(build_log_artifact.provider.retrieve!(artifact: build_log_artifact))
+      else
+        json_error!(
+          error_message: "Logs file missing for build #{build_number}",
+          error_key: "Build.LogsMissing",
+          error_code: 404
+        )
+      end
+
+      plain_artifact_file_content = convert_ansi_to_plain_text(artifact_file_content)
+      log_array = plain_artifact_file_content.split("\n").collect do |log_line|
+        {
+          message: log_line
```

Comment:
This notation is here because in the future we'll also transfer the "metadata" of each row right? (e.g. warning, error). And we just don't have this information for now as we just store the plaintext log output for the time being

Reply:
Yeah, we'll have to talk about how we want to structure logs are some point. I think it would be OK to output plain text until the worker stuff is done. Don't want to do the log work twice.